addons/io/org.openhab.io.hueemulation/src/main/java/org/openhab/io/hueemulation/internal/HueEmulationServlet.java

```diff
@@ -0,0 +1,521 @@
+        return devices;
+    }
+
+    /**
+     * Returns the item name and voice name of each item
+     *
+     * @param username
+     * @return
+     *         Map<item name, item voice tag>
+     */
+    public Map<String, String> getHueDeviceNames() {
+        Collection<Item> items = getTagedItems();
+        Map<String, String> devices = new HashMap<String, String>();
+        Iterator<Item> it = items.iterator();
+        while (it.hasNext()) {
+            Item item = it.next();
+            devices.put(item.getName(), item.getLabel());
+        }
+        return devices;
+    }
+
+    /**
+     * Converts an VoiceItem to a HueDevice
+     *
+     * @param voiceItem
+     * @return
+     *         HueDevice
+     */
+    private HueDevice itemToDevice(Item item) {
+        State itemState = item.getState();
+        short bri = 0;
+        if (itemState instanceof DecimalType) {
+            bri = (short) ((((DecimalType) itemState).intValue() * 255) / 100);
+        } else if (itemState instanceof OnOffType) {
+            bri = (short) (((OnOffType) itemState) == OnOffType.ON ? 255 : 0);
+        }
+        HueState hueState = new HueState(bri > 0, bri);
+        HueDevice d = new HueDevice(hueState, item.getLabel(), item.getName());
+        return d;
+    }
+
+    /**
+     * Gets all items that match our tag
+     *
+     * @return
+     */
+    private Collection<Item> getTagedItems() {
+        Collection<Item> items = new LinkedList<Item>();
+        for (Item item : itemRegistry.getItems()) {
+            for (String tag : item.getTags()) {
+                if (tag.indexOf(HOMEKIT_PREFIX) == 0) {
+                    items.add(item);
+                    break;
+                }
+            }
+        }
+        return items;
+    }
+
+    /**
+     * Checks if the username exists in our user file
+     *
+     * @param userName
+     * @return
+     * @throws IOException
+     */
+    private boolean authorizeUser(String userName) throws IOException {
+        synchronized (USER_FILE) {
+            if (USER_FILE.exists()) {
+                List<String> userNames = IOUtils.readLines(new FileInputStream(USER_FILE));
+                return userNames.contains(userName);
+            }
+            return false;
+        }
+    }
+
+    /**
+     * Adds a username to the user file
+     *
+     * @param userName
+     * @throws IOException
+     */
+    private void addUser(String userName) throws IOException {
+        synchronized (USER_FILE) {
+            List<String> userNames;
+            if (USER_FILE.exists()) {
+                userNames = IOUtils.readLines(new FileInputStream(USER_FILE));
+            } else {
+                USER_FILE.getParentFile().mkdirs();
+                userNames = new LinkedList<String>();
+            }
+            if (!userNames.contains(userName)) {
+                userNames.add(userName);
+                IOUtils.writeLines(userNames, null, new FileOutputStream(USER_FILE));
```

Comment:
Afaik, this does not close the output stream by itself. You should therefore catch potential IOExceptions here and do a `IOUtils.closeQuietly`. Alternatively, use directly the Java7 [try-with-resources](https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html) construct.